### PR TITLE
feat: extended cloudflare context

### PR DESCRIPTION
**Pull request by barisbora**
### 🔗 Linked issue

resolves https://github.com/unjs/nitro/issues/391

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Extending context object with cloudflare CF Properties and headers

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/entries/cloudflare.ts [hunk 1]
```diff
@@ -22,7 +25,11 @@ async function handleEvent (event: FetchEvent) {
     body = Buffer.from(await event.request.arrayBuffer())
   }
 
-  const r = await nitroApp.localCall({
```
Comment (pi0) [inline src/runtime/entries/cloudflare.ts:25]: We could simply extend `event.context.cf = event.cf` in line before i guess.

File: src/runtime/entries/cloudflare.ts [hunk 2]
```diff
@@ -60,3 +67,28 @@ const baseURLModifier = (request: Request) => {
 function normalizeOutgoingHeaders (headers: Record<string, string | string[] | undefined>) {
   return Object.entries(headers).map(([k, v]) => [k, Array.isArray(v) ? v.join(',') : v])
 }
+
+function toNodeListener (app: App, context: H3EventContext): NodeListener {
+  const toNodeHandle: NodeListener = async function (req, res) {
+    const event = createEvent(req, res)
+    event.context = { ...event.context, ...context }
```
Comment (danielroe) [inline src/runtime/entries/cloudflare.ts:74]: I think we might want to add CF context as `event.context.cloudflare` for better type support.

Comment (timhanlon) [reply src/runtime/entries/cloudflare.ts:74]:
```suggestion
    event.context = { ...event.context, cloudflare: context }
```

Comment (danielroe) [reply src/runtime/entries/cloudflare.ts:74]: Oops - didn't mean to edit. What about?

```suggestion
    event.context.cloudflare = context
```

Comment (barisbora) [reply src/runtime/entries/cloudflare.ts]: Thanks for feedback, implemented 🙌

File: src/runtime/entries/cloudflare.ts [hunk 3]
```diff
@@ -60,3 +67,28 @@ const baseURLModifier = (request: Request) => {
 function normalizeOutgoingHeaders (headers: Record<string, string | string[] | undefined>) {
   return Object.entries(headers).map(([k, v]) => [k, Array.isArray(v) ? v.join(',') : v])
 }
+
+function toNodeListener (app: App, context: H3EventContext): NodeListener {
+  const toNodeHandle: NodeListener = async function (req, res) {
+    const event = createEvent(req, res)
+    event.context.cloudflare = context
```
Comment (danielroe) [inline src/runtime/entries/cloudflare.ts:74]: Should this be:

```suggestion
    event.context.cloudflare = context.cf
```

Comment (timhanlon) [reply src/runtime/entries/cloudflare.ts:74]: I think the answer to this depends on where you'll want the [fetch event context](https://developers.cloudflare.com/workers/runtime-apis/fetch-event/#parameters) to live when module syntax support lands in #681.

**Comment by pi0** [comment]
Thanks for working on this @barisbora ❤️ Adding support with a slightly more generic approach in #927

**Pull request closed without merging**